shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -200,7 +200,7 @@ private JobAttempt newJob(AlluxioURI filePath, int replication) {
   private void waitJob() {
     AtomicBoolean removed = new AtomicBoolean(false);
     while (true) {
-      mSubmittedJobAttempts = mSubmittedJobAttempts.parallelStream().filter((jobAttempt) -> {
+      mSubmittedJobAttempts = mSubmittedJobAttempts.stream().filter((jobAttempt) -> {
```

Comment:
Does this method spin over checking jobs?

Reply:
This method spins over checking jobs until it removed one because one is complete/cancelled/failed.

Reply:
Can you make it parallel but put a magic sleep (100ms) at the end.

Reply:
if we can know if there is no pending task left after this, then we can even increase the sleep (to 500ms) but avoid when there are pending jobs.

Reply:
No that's not good. The whole point is that this mSubmittedJobAttempts is a running list of 1000 jobs. Once it removes a single one from this list, it escapes this loop to add more to fill 1000.

Reply:
Can you cache the result of `mClient.getJobStatus(mJobId)` in the job attempt whenever it's Completed/Failed?

Reply:
This whole structure seems strange? Could we not just have a threadpool/executor service that just runs tasks, one for each path? The generator just goes through the tree and adds tasks. It is strange that there is this spin loop that has to check all jobs?